packages/docs-markdown/src/extension.ts

```diff
@@ -201,8 +201,10 @@ export async function activate(context: ExtensionContext) {
 			}
 		}
 	});
-	workspace.onDidCloseTextDocument(e => {
-		metadataTreeProvider.refresh();
+	workspace.onDidChangeTextDocument(e => {
+		if (e.document.languageId === 'markdown') {
```

Comment:
Why did you change this? We want it to refresh and show the welcome text if the user closes the last Markdown file. Also if the user switches active focus to a non-Markdown file, we also want to refresh so we clear the tree (since it doesn't apply to the current editor).

Reply:
I believe that this still works the way we want it, I do see the welcome page. I'll remove the guard for `'markdown'` though.